form/src/main/java/org/dhis2/form/ui/FormViewFragmentFactory.kt

```diff
@@ -17,7 +18,8 @@ class FormViewFragmentFactory(
     private val completionListener: ((percentage: Float) -> Unit)?,
     private val onDataIntegrityCheck: ((result: DataIntegrityCheckResult) -> Unit)?,
     private val onFieldItemsRendered: ((fieldsEmpty: Boolean) -> Unit)?,
-    private val resultDialogUiProvider: EnrollmentResultDialogUiProvider?,
+    private val enrollmentResultDialogProvider: EnrollmentResultDialogProvider?,
+    private val eventCompletionDialogProvider: EventCompletionDialogProvider?,
```

Comment:
Do you think it could make sense to have an interface that implements both EnrollmentResultDialogProvider and EventCompletionDialogProvider with a method that returns Pair<BottomSheetDialogUiModel, List<FieldWithIssue>> and here we can use only that interface instead of two providers?

Reply:
mmm that was the original plan, but as the event dialog flow is so much more complex I ended up separating it, maybe now that it is all implemented I could merge them into one, I will give it a go and see how it goes

Reply:
Well, it took me a while and I had to retest it all but it is done!